idea/vara-eth/frontend/src/app/api/use-mirror-contract.ts

```diff
@@ -1,19 +1,24 @@
+import { useQuery } from '@tanstack/react-query';
 import { getMirrorClient } from '@vara-eth/api';
 import type { Address } from 'viem';
-import { useAccount } from 'wagmi';
-import { useWalletClient, usePublicClient } from 'wagmi';
+import { useAccount, usePublicClient, useWalletClient } from 'wagmi';
 
 const useMirrorContract = (mirrorAddress: Address) => {
   const ethAccount = useAccount();
   const { data: walletClient } = useWalletClient({ chainId: ethAccount.chain?.id });
   const publicClient = usePublicClient({ chainId: ethAccount.chain?.id });
 
-  if (!walletClient || !publicClient) {
-    return { mirrorContract: null, isLoading: true };
-  }
+  return useQuery({
```

Comment:
Same comments as above - `queryKey` should be as broad as possible, `queryFn` should return only awaited result - everything else is an error.